src/factory.js

```diff
@@ -104,19 +105,19 @@ class Factory {
    * @returns {Promise<ControllerDaemon>}
    */
   async spawn (options = { }) {
+    const type = options.type || this.opts.type
     const opts = merge(
-      this.overrides[options.type || this.opts.type],
+      this.overrides[type],
+      // conditionally include ipfs based on which type of daemon we will spawn when none has been specifed
+      (type === 'js' || type === 'proc') ? {
```

Comment:
What about when the user does:

```
factory.spawn({
  'type': 'js',
  ipfsModule: {
    path: require.resolve('some/other/path/to/ipfs'),
    ref: require('some/other/path/to/ipfs'),
  }
})
```

I think with this change the `ipfsModule` option the user passes in will get ignored

Reply:
no, user options is the last arg and overrides everything above

Reply:
Yeah, just thought of that :)

Reply:
It still evaluates `require('ipfs')` which does not work when used in js-ipfs.

Reply:
This is fixed in https://github.com/ipfs/js-ipfsd-ctl/pull/450